lib/truffle/truffle/cext_ruby.rb

```diff
@@ -43,7 +43,8 @@ def rb_define_method(mod, name, function, argc)
       res
     end
 
-    mod.define_method(name, method_body)
+    method_body_with_arity = Primitive.proc_specify_arity(method_body, argc)
```

Comment:
Here you should pass `-1` if `argc` is `-2`, because with `-2` it's still arity number -1 (any nb of args) and not arity number -2 (at least 1 arg + any nb of args).